manual_hook.rb

```diff
@@ -2,56 +2,75 @@
 
 require 'resolv'
 
-def setup_dns(domain, txt_challenge)
-  resolved = false;
-  singleLoop = false;
-  dns = Resolv::DNS.new;
-  acme_domain = "_acme-challenge."+domain; 
-  puts "Checking for pre-existing TXT record for the domain: \'#{acme_domain}\'."
-
-  until resolved
-    dns.each_resource(acme_domain, Resolv::DNS::Resource::IN::TXT) { |resp|
-     if resp.strings[0] == txt_challenge
-       puts "Found #{resp.strings[0]}. match."
-       resolved = true;
-     else
-       puts "Found #{resp.strings[0]}. no match."
-     end
-    }
-
-    if !resolved
-     if !singleLoop
-       puts "Create TXT record for the domain: \'#{acme_domain}\'. TXT record:"
-       puts "\'#{txt_challenge}\'"
-       puts "Press enter when DNS has been updated..."
-       $stdin.readline()
-       singleLoop = true
-     end
-
-     puts "Didn't find a match for #{txt_challenge}"; 
-     puts "Waiting to retry..."; 
-     sleep 30; 
+# Struct to store a challenge
+Challenge = Struct.new(:domain, :acme_domain, :txt_challenge)
+
+
+# Check if a challenge is resolved, returns true in that case
+def resolved?(dns, challenge)
+  dns.each_resource(challenge[:acme_domain], Resolv::DNS::Resource::IN::TXT) { |resp|
+    if resp.strings[0] == challenge[:txt_challenge]
+      puts "Found #{resp.strings[0]}. match."
+      return true
+    else
```

Comment:
remove per @rudiedirkx. not sure why?

Reply:
I think you are right with https://github.com/jbjonesjr/letsencrypt-manual-hook/issues/10#issuecomment-390383142 the else shouldn't be dropped but we have to iterate over resp.strings

Reply:
Taking a closer look, L11 already iterates over all TXT records for that domain. We only have to remive the return in the else branch.

Reply:
So what I was originally conveying with this else is that I found a DNS txt record for the site (`_acme_challenge.blahblah.com`), but the value of the TXT record was not valid. This is a different error case than if no TXT record was not found at all.

Maybe this is not relevant, but helps if someone mistyped the value, and is instead trying to figure out "why the TXT record never propagated to the LE servers"

I'll update the error messages to be more clear, but leave the `else` in place.